build-vm-qemu

```diff
@@ -212,9 +212,8 @@ vm_kill_qemu() {
 
 vm_fixup_qemu() {
     vm_fixup_kvm
-
-    case $qemu_device in
-        virtio*)
+    case $BUILD_HOST_ARCH in
+        armv6l|armv7l|armv8l|aarch32|aarch64|aarch64_ilp32|ppc|ppcle|ppc64|ppc64le|riscv64|s390|s390x)
```

Comment:
why is this not including hppa and the other architectures where `vm_startup_qemu()` is setting `qemu_device` ? what is the reason for changing away from that detection to the build host? shouldn't the setting of qemu_device then removed as this is the only place where it was referenced?

Reply:
answering my own question, I see the problem is that at the time of execution of `vm_fixup_qemu()` the `$qemu_device` isn't set already based on BUILD_HOST_ARCH, as it is only set at vm_startup_qemu times, where it is much too late and breaking all the other subtle logic in build-vm-km.